### PR TITLE
docker/podman: tag crimson flavor container as 'sha-1-flavor'

### PR DESCRIPTION
**Pull request by ideepika**
see also: https://github.com/ceph/ceph-container/pull/1704/

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -1161,8 +1161,16 @@ def task(ctx, config):
...
+        # right now we only have support for centos 8 images
+        os = config.get('os_type')
+        os_ver = config.get('os_version')
+        if 'flavor' not in config:
+            config['flavor'] = 'default'
+        flavor = config.get('flavor')
+
         if sha1:
-            ctx.ceph[cluster_name].image = container_image_name + ':' + sha1
+            #eg: quay.ceph.io/ceph-ci/ceph:8f1620b3c9322a518ba8b8c6573435d81b9c0756:centos-8:default
```
Comment (sebastian-philipp) [inline qa/tasks/cephadm.py:1172]: this should look like so:

```suggestion
            #eg: quay.ceph.io/ceph-ci/ceph:8f1620b3c9322a518ba8b8c6573435d81b9c0756-centos-8-default
```

IMO we shouldn't use colons in the image tag itself.

File: qa/tasks/cephadm.py [hunk 2]
```diff
@@ -1161,8 +1161,16 @@ def task(ctx, config):
...
+        os = config.get('os_type')
+        os_ver = config.get('os_version')
+        if 'flavor' not in config:
+            config['flavor'] = 'default'
+        flavor = config.get('flavor')
+
         if sha1:
-            ctx.ceph[cluster_name].image = container_image_name + ':' + sha1
+            #eg: quay.ceph.io/ceph-ci/ceph:8f1620b3c9322a518ba8b8c6573435d81b9c0756-centos-8-default
+            ctx.ceph[cluster_name].image = container_image_name + ':' + sha1 + '-' + os_type + '-' + os_ver + ':' + flavor
```
Comment (sebastian-philipp) [inline qa/tasks/cephadm.py:1173]:
```suggestion
            ctx.ceph[cluster_name].image = container_image_name + ':' + sha1 + '-' + os_type + '-' + os_ver + '-' + flavor
```

File: qa/tasks/cephadm.py [hunk 3]
```diff
@@ -1161,8 +1161,16 @@ def task(ctx, config):
     ref = None
     if not ctx.ceph[cluster_name].image:
         sha1 = config.get('sha1')
+        # right now we only have support for centos 8 images
+        os = config.get('os_type')
+        os_ver = config.get('os_version')
+        if 'flavor' not in config:
+            config['flavor'] = 'default'
+        flavor = config.get('flavor')
```
Comment (tchaikov) [inline qa/tasks/cephadm.py:1169]:
```suggestion
        flavor = config.get('flavor', 'default')
```

**Comment by tchaikov** [comment]
might want to revise the commit title 

> qa/cephadm: change docker container format

1. we could use docker or podman.
1. you are not changing the container format
1. you are changing the naming scheme of the tag of ceph images.

**Comment by ideepika** [comment]
@tchaikov adopted your suggestions, thanks for reviewing

**Comment by tchaikov** [comment]
@sebastian-philipp could you help re-review this change? i think it's ready to merge once https://github.com/ceph/ceph-container/pull/1704 is merged.

**Pull request merged**